cmake/configure_lvi_mitigation_build.cmake

```diff
@@ -1,25 +1,81 @@
 # Copyright (c) Open Enclave SDK contributors.
 # Licensed under the MIT License.
 
+macro(detect_compiler BINDIR CC)
+    # Compiler detection logic:
+    # 1. Use either the default compiler or the one specified via CC/CXX.
+    # 2. If the compiler in 1. does not exist and equals to clang/clang++,
+    #    the logic tries to pick the clang/clang++ with specific versions (order: 9, 8, 7).
```

Comment:
I think if someone specifies a compiler with CC that doesn't exist we should just not worry about it and fail. That is, blindly use CC/CXX no matter what they are.